packages/types/mongoose-store/usage.ts

```diff
@@ -0,0 +1,63 @@
+import * as mongoose from 'mongoose';
+
+import MongooseStore, { Store } from './index';
+
+interface User {
+  id: string;
+  name: string;
+  age: number;
+  todos: Todos[];
+  getFullName: () => string;
+}
+
+interface Todos {
+  id: string;
+  what: string;
+  deadline: Date;
+  userId: string;
+  pastDeadline: (when: Date) => boolean;
+}
+
+describe('Store Interface', function() {
+  const store: Store<User> = null; // = new MongooseStore<User>(mongoose.createConnection('localhost'));
+
+  it('should showcase a chainable query interface', function() {
+    // Fluent interface example
+    return store.query()
+      .where('name', 'jon')
+      .gt('age', 18)
+      // .gt('age', 'foo') => Argument of type '"foo"' is not assignable to parameter of type 'number'.
+      .or(store.or().where('name', 'snow'))
+      .getSingle().tap((u) => console.log(u.age));
+  });
+  it('should showcase a object-based query interface', function() {
+    // Mongoose-style large query object example
+    return store.find({
+      where: {
+        eq: { name: 'jon' },
+        gt: { age: 18 },
+        or: {
+          eq: { name: 'snow' }
+        }
+      }
+    });
+  });
```

Comment:
These two are alternative proposals.
Note that all of the params are type-checked against the original models.

I still need to figure out how to prevent adding `Function` properties though.